### DISCUSSION TITLE
Send pdf to printer

### DISCUSSION THREAD

**Discussion by ange3791** [OpenPrinting/cups-filters#461] [General]
I have an older non IPP printer that can print postscript or pdf files sent raw. I have the manufacturer's PPD file. How can I set up my queue using the ppd to send PDFs to the printer? 

Thanks,
Andy

**Comment by pdfkungfoo** [comment]
Maybe by adding one or more `*cupsFilter:`-lines to your PPD somewhere within the first 10 lines? Something like:
~~~~bash
*cupsFilter: "application/pdf 0 -"
*cupsFilter: "application/vnd.cups-pdf 0 -"
*cupsFilter: "application/vnd.cups-postscript 0 -"
~~~~

The third line allows CUPS to also send PostScript

Or use the more modern CUPS syntax:

~~~~bash
*cupsFilter2: "application/vnd.cups-pdf application/vnd.cups-pdf 10 -"
~~~~
